src/main/kotlin/com/jetbrains/snakecharm/lang/psi/types/SmkPepConfigType.kt

```diff
@@ -0,0 +1,101 @@
+import com.jetbrains.python.psi.AccessDirection
+import com.jetbrains.python.psi.PyExpression
+import com.jetbrains.python.psi.impl.ResolveResultList
+import com.jetbrains.python.psi.resolve.PyResolveContext
+import com.jetbrains.python.psi.resolve.RatedResolveResult
+import com.jetbrains.python.psi.types.PyType
+import com.jetbrains.snakecharm.codeInsight.resolve.SmkResolveUtil
+import com.jetbrains.snakecharm.lang.psi.SmkFile
+import com.jetbrains.snakecharm.lang.psi.impl.SmkPsiUtil
+import kotlinx.serialization.modules.EmptySerializersModule
+import org.jetbrains.yaml.psi.YAMLFile
+import org.jetbrains.yaml.psi.YAMLScalarText
+import org.jetbrains.yaml.psi.impl.YAMLMappingImpl
+import org.jetbrains.yaml.psi.impl.YAMLPlainTextImpl
+
+class SmkPepConfigType(smkFile: SmkFile) : PyType {
+    override fun resolveMember(
+        name: String,
+        location: PyExpression?,
+        direction: AccessDirection,
+        resolveContext: PyResolveContext
+    ): List<RatedResolveResult> {
+        if (!SmkPsiUtil.isInsideSnakemakeOrSmkSLFile(location)) {
+            return emptyList()
+        }
+        val resolveResult = ResolveResultList()
+        sectionArgs
+            .filter { it.second == name }
+            .forEach {
+                resolveResult.poke(it.first, SmkResolveUtil.RATE_NORMAL)
+            }
+        return resolveResult
+    }
+
+    override fun getCompletionVariants(
+        completionPrefix: String?,
+        location: PsiElement?,
+        context: ProcessingContext?
+    ): Array<LookupElement> = sectionArgs.map {
+        LookupElementBuilder
+            .create(it.second)
+    }.toTypedArray()
+
+    private val sectionArgs = getYamlKeys(getYamlFile(smkFile))
+
+    override fun getName(): String = "pep.config"
+
+    override fun isBuiltin(): Boolean = false
+
+    override fun assertValid(message: String?) {
+        sectionArgs.forEach {
+            if (!it.first.isValid) {
+                throw PsiInvalidElementAccessException(it.first, message)
+            }
+        }
+    }
+
+    companion object {
+        private fun getYamlKeys(yamlFile: PsiFile?): List<Pair<PsiElement, String>> {
+            if (yamlFile == null) return emptyList()
+            val ymlFile = yamlFile as YAMLFile
+            val completionList = mutableListOf<Pair<PsiElement, String>>()
+            ymlFile.documents.forEach { document ->
+                document.topLevelValue?.children?.asSequence()?.filter { child ->
+                    // We can't refer to foo: boo: key
+                    child.lastChild !is YAMLMappingImpl
+                }?.forEach { child ->
+                    completionList.add(
+                        // In Yaml file can be key like {foo.boo: }, we can refer to this section
+                        // like pep.config.foo, but not pep.config.foo.boo
+                        child.firstChild to
+                                child.firstChild.text.substringBefore('.')
+                    )
```

Comment:
Not all items are valid for completion list, e.g. `key1:key2` is invalid option, because `pep.config.key1:key2` is a syntax error. Make code smarter and add to completion list only valid identifiers